website/static/js/share/results.js

```diff
@@ -1,126 +1,272 @@
+var Description = {
+    controller: function(vm) {
+        var self = this;
+        self.showAll = false;
+    },
+    view: function(ctrl, params) {
+        var result = params.result;
+        if ((result.description || '').length > 350) {
+            return m('p.readable.pointer', {
+                onclick: function() {
+                    ctrl.showAll = !ctrl.showAll;
+                    }
+                },
+                ctrl.showAll ? result.description : $.truncate(result.description, {length: 350})
+            );
+        } else {
+            return m('p.readable', result.description);
+        }
+    }
+};
+
+var Contributors = {
+    controller: function(vm) {
+        var self = this;
+        self.showAll = false;
+    },
+    view: function(ctrl, params) {
+        var result = params.result;
+        var contributorViews = $.map(result.contributors, function(contributor, i) {
+                return m.component(Contributor, $.extend({contributor: contributor, index: i}, params));
+            });
+
+        return m('span.pull-left', {style: {'text-align': 'left'}},
+            ctrl.showAll || result.contributors.length < 8 ?
+                contributorViews :
+                m('span', [
+                    contributorViews.slice(0, 7),
+                    m('br'),
+                    m('a', {onclick: function(){ctrl.showAll = !ctrl.showAll;}}, 'See All')
+                ])
+        );
+
+    }
+};
+
+var Contributor = {
+    view: function(ctrl, params) {
+        var contributor = params.contributor;
+        var index = params.index;
+        var vm = params.vm;
+        return m('span', [
+            m('span', index !== 0 ? ' · ' : ''),
+            m('a', {
+                onclick: function() {
+                    utils.updateFilter(vm, 'match:contributors.familyName:' + contributor.familyName, true);
+                    utils.updateFilter(vm, 'match:contributors.givenName:' + contributor.givenName, true);
+                }
+            }, contributor.name)
+        ]);
+    }
+};
+
+var Subjects = {
+    controller: function(vm) {
+        var self = this;
+        self.showAll = false;
+    },
+    view: function(ctrl, params){
+        var result = params.result;
+        var subjectViews = $.map(result.subjects || [], function(subject, i) {
+            return m.component(Subject, $.extend({subject: subject}, params));
+        });
+        if (ctrl.showAll || (result.subjects || []).length <= 5) {
+            return m('span', subjectViews);
+        }
+        return m('span', [
+            subjectViews.slice(0, 5),
+            m('br'),
+            m('div', m('a', {onclick: function() {ctrl.showAll = !ctrl.showAll;}},'See All'))
+        ]);
+
+    }
+};
+
+var Subject = {
+    view: function(ctrl, params) {
+        var subject = params.subject;
+        var vm = params.vm;
+        return m('span', m('.badge.pointer', {onclick: function(){
```

Comment:
Clickable `span` and `div` tags present a number of accessibility problems because they can't be focused and activated without a mouse; use `button` or `a` with `href="#"` (probably the better option here to get the styling right) instead.